youtube_dl/downloader/external.py

```diff
@@ -192,15 +195,19 @@ def supports(cls, info_dict):
     def available(cls):
         return FFmpegPostProcessor().available
 
+    def get_execname(self):
+        return self.execname
```

Comment:
> 2. It should not store state.

How could it work otherwise? `_call_downloader()` instantiates an instance of `FFmpegPostProcessor`, and only at that time is a decision made on the executable name.

> 3. It should be implemented similar to available.

Meaning as an `@property`? OK.